tim1/src/main/java/isamrs/tim1/controller/BranchOfficeController.java

```diff
@@ -37,8 +44,14 @@
 		}
 		BranchOffice bo = new BranchOffice();
 		bo.setId(null);
-		bo.setLocation(branch.getLocation());
-		bo.setName(bo.getName());
+		Location l;
```

Comment:
Better to revert to old code (using branch.getLocation(), not searching if there exists Location with these coordinates).